.github/workflows/pull-request-workflow.yml

```
@@ -0,0 +1,41 @@
+name: pull-request-checks
+run-name: ${{ github.triggering_actor }} has started mandatory checks
+
+on: [pull_request]
+
+env:
+  DEVELOPER_DIR: /Applications/Xcode_15.4.app/Contents/Developer
+
+jobs:
+    run-tests:
+        runs-on: macos-14
+        steps:
+            - name: checkout-repo
+              uses: actions/checkout@v4
+
+            - name: testing
+              working-directory: ./LifoContainers
+              run: |
+                xcodebuild \
+                  -project LifoContainers.xcodeproj \
+                  -scheme LifoContainersTests \
+                  test
+
+    run-linters:
+        runs-on: macos-14
+        steps:
+            - name: checkout-repo
+              uses: actions/checkout@v4
+
+            - name: update-brew
+              run: brew update
+
+            - name: install-linters
+              run: |
+                brew install swiftformat
+                brew install swiftlint
+
+            - name: linting
+              run: |
+                chmod +x ./scripts/run-linters.sh
+                sh ./scripts/run-linters.sh
```

I'd suggest splitting this into two separate steps because you are changing permissions for your script.

```suggestion
            - name: Add execution permission to the run-linters.sh
              run: chmod +x ./scripts/run-linters.sh
            - name: Linting
              run: sh ./scripts/run-linters.sh
```

---

I don't really think adding permissions to execute the script is a different step cause these two commands are super connected to each other.

---

I'm concerned that your step is named `Linting`, but inside it, you're changing permissions for your script and linting the sources. I'd suggest using my example or changing the step name to the `Set Permissions and Linting` to explain that you are changing permissions too.

---

The step has been renamed